### ISSUE TITLE
Need to be able to send secure session cookie for localhost

### ISSUE THREAD

**Issue by jrmcgarvey** [rack/rack-session#40]
Right now, unless it is ssl, you can't send a secure session cookie.  It is perfectly valid to do this for localhost, and very desirable for development.

In lib/rack/session/abstract/id.rb, I think it should say something like:

  def security_matches?(request,options)
    return true unless options[:secure]
    request.ssl? || request.host == "localhost"
  end

**Comment by ioquatix** [comment]
That seems reasonable to me, but I don't fully grok the implications. Do you mind submitting a PR? Then we can discuss the specifics.

**Comment by jrmcgarvey** [comment]
Yes, this is documented [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies).  RFC 6265 punts to the browser implementation on the question of what constitutes a secure connection.  See my response on the PR.

**Comment by ioquatix** [comment]
Just for reference, I assume we are talking about:

> A cookie with the Secure attribute is only sent to the server with an encrypted request over the HTTPS protocol. It's never sent with unsecured HTTP (except on localhost), which means [man-in-the-middle](https://developer.mozilla.org/en-US/docs/Glossary/MitM) attackers can't access it easily. Insecure sites (with http: in the URL) can't set cookies with the Secure attribute. However, don't assume that Secure prevents all access to sensitive information in cookies. For example, someone with access to the client's hard disk (or JavaScript if the HttpOnly attribute isn't set) can read and modify the information.

**Comment by jrmcgarvey** [comment]
Yes, that's the only reference I can find on the issue, other than the RFC.
